dbms/src/Encryption/KeyspacesKeyManager.h

```diff
@@ -62,7 +58,7 @@ class KeyspacesKeyManager : public KeyManager
 public:
 #endif
     ProxyHelper * proxy_helper;
-    UniversalPageStoragePtr ps_write = nullptr;
+    std::weak_ptr<UniversalPageStorage> ps_write;
```

Comment:
Add some comments about why we need a weak_ptr here

Reply:
done